src/App.tsx

```diff
@@ -8,7 +12,76 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+
+type Callback = (newQuery: string) => void;
+
+function debaunce(f: Callback, delay: number) {
+  let timerId: number;
+
+  return (...args: []) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+}
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoId, setTodoid] = useState(0);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [select, setSelect] = useState<TodoStatus>(TodoStatus.ALL);
```

Comment:
it's not clear what should be stored in select variable, think about variable name